### PR TITLE
CommonSenseQA

### PR DESCRIPTION
**Pull request by domrigoglioso**
**Patch description**
Adds [CommonSenseQA dataset](https://www.tau-nlp.org/commonsenseqa) 

**Logs**
```
python3 examples/display_data.py -t commonsenseqa -v
[creating task(s): commonsenseqa]
[id]: commonsenseqa
[text]: The sanctions against the school were a punishing blow, and they seemed to what the efforts the school had made to change?
[label_candidates]: ignore|enforce|authoritarian|yell at|avoid
[labels]: ignore
- - - - - - - END OF EPISODE - - - - - - - - - -
~~
[id]: commonsenseqa
[text]: Sammy wanted to go to where the people were.  Where might he go?
[label_candidates]: race track|populated areas|the desert|apartment|roadblock
[labels]: populated areas
- - - - - - - END OF EPISODE - - - - - - - - - -
~~
[id]: commonsenseqa
[text]: To locate a choker not located in a jewelry box or boutique where would you go?
[label_candidates]: jewelry store|neck|jewlery box|jewelry box|boutique
[labels]: jewelry store
- - - - - - - END OF EPISODE - - - - - - - - - -

```

**Data tests (if applicable)**
```
python3 tests/datatests/test_new_tasks.py 
.
----------------------------------------------------------------------
Ran 1 test in 8.341s

OK
```

### CODE REVIEW

File: parlai/tasks/commonsenseqa/agents.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            dpath = os.path.join(jsons_path, 'train.jsonl')
+        else:
+            raise ValueError('Datatype not train, test, or valid')
+        episodes = []
+        with open(dpath) as f:
+            for line in f:
+                episodes.append(json.loads(line))
+        if dtype.startswith('valid'):
+            episodes = episodes[len(episodes) // 2 :]
+        elif dtype.startswith('test'):
```
Comment (klshuster) [inline parlai/tasks/commonsenseqa/agents.py:51]: is this how the dataset is setup? It might be good to `warn_once` that there is no available test set, rather than splitting up the validation set (unless the second half of the dev set _is_ the test set)
